packages/material-ui/src/Chip/Chip.js

```diff
@@ -38,182 +115,211 @@ export const styles = (theme) => {
-    avatar: {},
-    /* Styles applied to the avatar element if `size="small"`. */
-    avatarSmall: {},
-    /* Styles applied to the avatar element if `color="primary"`. */
-    avatarColorPrimary: {},
-    /* Styles applied to the avatar element if `color="secondary"`. */
-    avatarColorSecondary: {},
-    /* Styles applied to the icon element. */
-    icon: {
-      color: theme.palette.mode === 'light' ? theme.palette.grey[700] : theme.palette.grey[300],
-      marginLeft: 5,
-      marginRight: -6,
-    },
-    /* Styles applied to the icon element if `size="small"`. */
-    iconSmall: {
-      fontSize: 18,
-      marginLeft: 4,
-      marginRight: -4,
-    },
-    /* Styles applied to the icon element if `color="primary"`. */
-    iconColorPrimary: {
-      color: 'inherit',
-    },
-    /* Styles applied to the icon element if `color="secondary"`. */
-    iconColorSecondary: {
-      color: 'inherit',
-    },
-    /* Styles applied to the label `span` element. */
-    label: {
-      overflow: 'hidden',
-      textOverflow: 'ellipsis',
-      paddingLeft: 12,
-      paddingRight: 12,
-      whiteSpace: 'nowrap',
-    },
-    /* Styles applied to the label `span` element if `size="small"`. */
-    labelSmall: {
-      paddingLeft: 8,
-      paddingRight: 8,
-    },
-    /* Styles applied to the deleteIcon element. */
-    deleteIcon: {
+      /* Styles applied to the root element if `size="small"`. */
+      ...(styleProps.size === 'small' && {
+        height: 24,
+      }),
+      /* Styles applied to the root element if `color="primary"`. */
+      ...(styleProps.color === 'primary' && {
+        backgroundColor: theme.palette.primary.main,
+        color: theme.palette.primary.contrastText,
+      }),
+      /* Styles applied to the root element if `color="secondary"`. */
+      ...(styleProps.color === 'secondary' && {
+        backgroundColor: theme.palette.secondary.main,
+        color: theme.palette.secondary.contrastText,
+      }),
+      /* Styles applied to the root element if `onDelete` is defined. */
+      ...(styleProps.onDelete && {
+        '&.Mui-focusVisible': {
+          backgroundColor: emphasize(backgroundColor, 0.08),
+        },
+      }),
+      /* Styles applied to the root element if `onDelete` and `color="primary"` is defined. */
+      ...(styleProps.onDelete &&
+        styleProps.color === 'primary' && {
+          '&.Mui-focusVisible': {
+            backgroundColor: emphasize(theme.palette.primary.main, 0.2),
+          },
+        }),
+      /* Styles applied to the root element if `onDelete` and `color="secondary"` is defined. */
+      ...(styleProps.onDelete &&
+        styleProps.color === 'secondary' && {
+          '&.Mui-focusVisible': {
+            backgroundColor: emphasize(theme.palette.secondary.main, 0.2),
+          },
+        }),
+      /* Styles applied to the root element if `variant="outlined"`. */
+      ...(styleProps.variant === 'outlined' && {
```

Comment:
Needs to be a separate callback as you would override completely the previous definition about the same classes.